tests/client.test.ts

```diff
@@ -1,27 +1,31 @@
-import { Connection, Keypair, LAMPORTS_PER_SOL, SystemProgram, sendAndConfirmTransaction, Transaction, PublicKey } from "@solana/web3.js";
+import { Connection, Keypair, LAMPORTS_PER_SOL, SystemProgram, sendAndConfirmTransaction, Transaction, PublicKey, TransactionInstruction, Signer } from "@solana/web3.js";
+import * as fs from "fs";
+import * as os from "os";
 
 jest.setTimeout(60000);
 
+const MEMO_PROGRAM_ID = new PublicKey("MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr");
+
 test('send and confirm transaction', async () => {
-    const connection = new Connection('http://127.0.0.1:8890', 'confirmed');
-    const payer = Keypair.generate();
-    const toAccount = Keypair.generate().publicKey;
+    const connection = new Connection('http://0.0.0.0:8890', 'confirmed');
 
-    const airdropSignature = await connection.requestAirdrop(payer.publicKey, LAMPORTS_PER_SOL * 2);
-    console.log('airdrop signature  ' + airdropSignature);
-    await connection.confirmTransaction(airdropSignature, 'finalized');
-    console.log('confirmed');
+    const keypair_file = fs.readFileSync(`${os.homedir}/.config/solana/id.json`, 'utf-8');
+    const keypair_array = Uint8Array.from(JSON.parse(keypair_file));
+    const payer = Keypair.fromSecretKey(keypair_array);
 
     const transaction = new Transaction();
 
-    // Add an instruction to execute
     transaction.add(
-        SystemProgram.transfer({
-            fromPubkey: payer.publicKey,
-            toPubkey: toAccount,
-            lamports: LAMPORTS_PER_SOL,
-        }),
+        new TransactionInstruction({
+            programId: MEMO_PROGRAM_ID,
+            keys: [],
+            data: Buffer.from("Hello")
```

Comment:
buffer needs to contain random data so you don't end up creating the same tx twice